packages/xchain-ethereum/src/client.ts

```diff
@@ -584,37 +569,38 @@ export default class Client extends BaseXChainClient implements XChainClient, Et
    * @returns {TransactionResponse} The transaction result.
    */
   approve = async ({
-    walletIndex = 0,
-    spender,
-    sender,
-    feeOptionKey,
+    contractAddress,
+    spenderAddress,
+    feeOptionKey = 'fastest',
     amount,
+    walletIndex = 0,
+    gasLimitFallback,
```

Comment:
if we change the method params i think we should rather allow the user to specify the gasPrice & gasLimit

if either are NOT supplied  ,they should be calculated using   estimateGasPrices() & estimateApprove()

Reply:
It's just on option to set a "fallback" value for `gasLimit` in case `estimateApprove` might fail (we had such case in interfaces in the past - see [discussion at Discord](https://discord.com/channels/838986635756044328/842252939149967382/850203681798357023)), but it's not to override the result of `estimateApprove`. If `estimateGasPrices` will fail, there is already a fallback included to use `getDefaultGasPrices` - no need to set it again.

Reply:
makes sense